pkg/plugins/generic/generic_plugin.go

```diff
@@ -213,7 +213,7 @@ func needRebootNode(state *sriovnetworkv1.SriovNetworkNodeState, loadVfioDriver
 		}
 	}
 
-	if utils.IsSwitchdevModeSpec(state.Spec) {
+	if utils.IsSwitchdevModeSpec(state.Spec) || utils.IsDpuConfigured(state) {
```

Comment:
Should we add a similar check in needDrainNode (line173)?

Reply:
I think we don't need the check at line 173 anymore. Since it was added with the MCO plugin and we have deprecated the MCO plugin. The MCO plugin is used to move a node to the MCP, so we don't need to drain/reboot the node by the config daemon. But now, as the config daemon is only responsible for updating the `sriov_config.json`, we shall use the generic plugin to drain/reboot the node.